makefile

```diff
@@ -84,6 +84,31 @@ robot2015:
 robot2015-prog:
 	$(call cmake_build_target_fw, robot2015-prog)
 
+GDB_PORT ?= 3333
+.INTERMEDIATE: build/robot2015-gdb.pid
+build/robot2015-gdb.pid:
+# this will cache sudo use without a password in the environment
+# so we won't enter the gdb server and skip past the password prompt.
+	@sudo echo "starting pyocd-gdbserver, logging to build/robot2015-gdb.log"
+# now we can refresh the sudo timeout and start up the gdb server
+	sudo -v && { sudo pyocd-gdbserver --allow-remote --port $(GDB_PORT) --reset-break \
+	--target lpc1768 -S -G -B 0 > build/robot2015-gdb.log 2>&1 & sudo echo $$! > $@; }
```

Comment:
@jjones646, I think we should be using the `>>` operator rather than `>` here.  The double caret one appends to the file's existing contents, while the single caret overwrites the previous contents.